components/plugin_marketplace/marketplace_list/navigation_row/navigation_row.jsx

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import PropTypes from 'prop-types';
+import {changeOpacity, makeStyleFromTheme} from 'mattermost-redux/utils/theme_utils';
+
+import NavigationButton from './navigation_button';
+
+export default class NavigationRow extends React.PureComponent {
+    static propTypes = {
+        page: PropTypes.number.isRequired,
+        total: PropTypes.number.isRequired,
+        maximumPerPage: PropTypes.number.isRequired,
+        onNextPageButtonClick: PropTypes.func.isRequired,
+        onPreviousPageButtonClick: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+    };
+
+    canShowNextButton = () => {
+        const {page, maximumPerPage, total} = this.props;
+        const pageStart = page * maximumPerPage;
+        const pageEnd = pageStart + maximumPerPage;
+
+        return total >= maximumPerPage && pageEnd < total;
```

Comment:
not going to block the review but was wondering if it might easier to check the following instead of doing this, what do you think @kosgrz? 🤔 

```
const {page, maximumPerPage, total} = this.props;
const totalPages = Math.trunc(total / maximumPerPage);

return totalPages > page;
```

Reply:
@ali-farooq0 it's a definitely better way. I'd like to update the PR if it's not a problem now :)

Reply:
@kosgrz sure go for it.

Reply:
@prapti heads up a small change around when to show the next button coming up -- apologies for the churn.

Reply:
@ali-farooq0 thanks! Done :+1: